From 82defb40d1d69ef5cf1454f45bbe36c320ce5a99 Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Fri, 24 May 2019 23:30:00 +0200
Subject: [PATCH 3/4] hidl: livedisplay: Move .displaymodedefault to
 /data/vendor/display

* By default hal_lineage_livedisplay_default does not
  have permissions to write to /data/misc.

Change-Id: I5599a5e6602ea326322328aeda1c6eb0411a5568
---
 lineagehw/hidl/livedisplay/DisplayModes.cpp                    | 2 +-
 .../lineage.livedisplay@2.0-service.samsung-exynos.rc          | 3 +++
 .../lineage.livedisplay@2.0-service.samsung-qcom.rc            | 3 +++
 3 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/lineagehw/hidl/livedisplay/DisplayModes.cpp b/lineagehw/hidl/livedisplay/DisplayModes.cpp
index 0202be6..113905e 100644
--- a/lineagehw/hidl/livedisplay/DisplayModes.cpp
+++ b/lineagehw/hidl/livedisplay/DisplayModes.cpp
@@ -29,7 +29,7 @@ namespace samsung {
 
 static constexpr const char* kModePath = "/sys/class/mdnie/mdnie/mode";
 static constexpr const char* kModeMaxPath = "/sys/class/mdnie/mdnie/mode_max";
-static constexpr const char* kDefaultPath = "/data/misc/.displaymodedefault";
+static constexpr const char* kDefaultPath = "/data/vendor/display/.displaymodedefault";
 
 const std::map<int32_t, std::string> DisplayModes::kModeMap = {
     {0, "Dynamic"},
diff --git a/lineagehw/hidl/livedisplay/lineage.livedisplay@2.0-service.samsung-exynos.rc b/lineagehw/hidl/livedisplay/lineage.livedisplay@2.0-service.samsung-exynos.rc
index cf26d1c..3fd15b5 100644
--- a/lineagehw/hidl/livedisplay/lineage.livedisplay@2.0-service.samsung-exynos.rc
+++ b/lineagehw/hidl/livedisplay/lineage.livedisplay@2.0-service.samsung-exynos.rc
@@ -1,3 +1,6 @@
+on post-fs-data
+    mkdir /data/vendor/display 0770 system system
+
 service livedisplay-hal-2-0-samsung-exynos /system/bin/hw/lineage.livedisplay@2.0-service.samsung-exynos
     class hal
     user system
diff --git a/lineagehw/hidl/livedisplay/lineage.livedisplay@2.0-service.samsung-qcom.rc b/lineagehw/hidl/livedisplay/lineage.livedisplay@2.0-service.samsung-qcom.rc
index bd3ca6c..48ebd94 100644
--- a/lineagehw/hidl/livedisplay/lineage.livedisplay@2.0-service.samsung-qcom.rc
+++ b/lineagehw/hidl/livedisplay/lineage.livedisplay@2.0-service.samsung-qcom.rc
@@ -1,3 +1,6 @@
+on post-fs-data
+    mkdir /data/vendor/display 0770 system system
+
 service livedisplay-hal-2-0-samsung-qcom /system/bin/hw/lineage.livedisplay@2.0-service.samsung-qcom
     class late_start
     user system
-- 
2.17.1

